05/app.js

```diff
@@ -10,11 +10,68 @@ const stats = {
 /* tutaj umieść swój kod */
 
 
+
+const pList = document.querySelectorAll('p');
+const linkList = document.querySelectorAll('a');
+
+pList.forEach(p => {
+    const paragraphValue = p.dataset.id;
+    
+    if (typeof paragraphValue !== 'undefined') {
+        stats.paragraphs[paragraphValue] = 0;
+    }
+    
+})
+
+linkList.forEach(link => {
+    const linkValue = link.getAttribute('href');
+    
+    if (typeof linkValue !== 'undefined') {
+        stats.links[linkValue] = 0;
+    }
+})
+
+const onClick = function (e) {
+    const el = e.target;
+    if (el.hasAttributes('href')) {
+        e.preventDefault();
+    }
+    
+    if(el.dataset.id === 'p1') {
+        ++stats.paragraphs.p1;
+    }
```

Comment:
To rozwiązanie jest o tyle problematyczne, że mając 100 różnych "opcji" będzie 100 ifów.
Dodawanie kolejnych to również kolejne zmiany.

W JS możemy identyfikować właściwości na podstawie wartości w zmiennej tj.

```
const person = {name: 'Mateusz'}
const property = 'name';

console.log(person[property]); // => Mateusz
```

Tutaj ten mechanizm warto zastosować czyli 
```
const propName = el.dataset.id;
++stats.paragraphs[propName];
```

Trzeba jeszcze pamiętać, o tym, aby sprawdzić czy istnieje taki element. Jeśli nie to go utowrzyć tj. `if(typeof stats.paragraphs[propName] === 'undefined') { stats.paragraphs[propName] = 0; }`